sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -251,6 +251,24 @@ abstract class JdbcDialect extends Serializable with Logging{
           s"${this.getClass.getSimpleName} does not support function: $funcName")
       }
     }
+
+    override def visitOverlay(inputs: Array[String]): String = {
+      if (isSupportedFunction("OVERLAY")) {
+        super.visitOverlay(inputs)
+      } else {
+        throw new UnsupportedOperationException(
+          s"${this.getClass.getSimpleName} does not support function: OVERLAY")
+      }
+    }
+
+    override def visitTrim(direction: String, inputs: Array[String]): String = {
```

Comment:
With the current framework, I don't think `OVERLAY` and `TRIM` are SQL functions (`visitSQLFunction`). I think `TRIM`  is common enough, but `OVERLAY` is not. Maybe we should by default throw error in `JDBCSQLBuilder`. H2 dialect can override `compileExpression` to support OVERLAY.